corehq/apps/reports/tasks.py

```diff
@@ -266,26 +235,49 @@ def _format_filename(form_info, question_id, extension, case_id_to_name):
     return filename
 
 
-def _write_attachments_to_file(fpath, use_transfer, num_forms, forms_info, case_id_to_name):
-
-    if not (os.path.isfile(fpath) and use_transfer):  # Don't rebuild the file if it is already there
-        with open(fpath, 'wb') as zfile:
-            with zipfile.ZipFile(zfile, 'w') as multimedia_zipfile:
-                for form_number, form_info in enumerate(forms_info):
-                    form = form_info['form']
-                    for attachment in form_info['attachments']:
-                        filename = _format_filename(
-                            form_info,
-                            attachment['question_id'],
-                            attachment['extension'],
-                            case_id_to_name
-                        )
-                        zip_info = zipfile.ZipInfo(filename, attachment['timestamp'])
-                        multimedia_zipfile.writestr(zip_info, form.get_attachment(
-                            attachment['name']),
-                            zipfile.ZIP_STORED
-                        )
-                    DownloadBase.set_progress(build_form_multimedia_zip, form_number + 1, num_forms)
+def _write_attachments_to_file(fpath, num_forms, forms_info, case_id_to_name):
+    total_size = 0
+    with open(fpath, 'wb') as zfile:
+        with zipfile.ZipFile(zfile, 'w') as multimedia_zipfile:
+            for form_number, form_info in enumerate(forms_info, 1):
+                form = form_info['form']
+                for attachment in form_info['attachments']:
+                    total_size += attachment['size']
+                    if total_size >= MAX_MULTIMEDIA_EXPORT_SIZE:
+                        raise Exception("Refusing to make multimedia export bigger than {} GB"
+                                        .format(MAX_MULTIMEDIA_EXPORT_SIZE / 1024**3))
```

Comment:
The convention is to use a unique exception class that can be caught without masking other exceptions. Example: `MaxFileSizeExceeded`. This could be especially handy in a test that asserts that the exception is raised at the proper time (probably while mocking `MAX_MULTIMEDIA_EXPORT_SIZE` to be reasonable/small for a test).

Reply:
I agree, if the exception is meant to be caught.  This intentionally fails hard to prevent a worse failure - it shouldn't be encountered in regular operation, so I don't think a custom exception adds value that the message doesn't.  Since I later added a proper up-front check, I was actually considering dropping this anyways.  Thoughts?

Reply:
If there was a test that ensured this code actually does what is expected then there would be code that calls it regularly and catches it... maybe I was being too subtle in asking for tests?

More generally, while nothing is catching it right now, raising `Exception` could encourage a future developer who _does_ have a reason to catch it to do something like `except Exception`, which would be a bad thing. I'll leave it up to your discretion if you think it's a case of YAGNI.

I wouldn't drop it. It's worth doing the check in two places since it's easy and ensures that this will fail if it happens to be executed on a code path that somehow omits the up-front check.

Reply:
> maybe I was being too subtle in asking for tests?

No I picked up on that - just a little challenging for this use case considering the pieces it needs.  Been a busy week with a lot of high prio stuff.  I'll see if I can get to it, but will merge this in the meantime.